File: app/src/main/java/stock/com/ui/pojo/WatchlistPojo.java
package stock.com.ui.pojo;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class WatchlistPojo extends BasePojo {
    @SerializedName("stock")
    @Expose
    public ArrayList<WatchStock> stock = null;

    public class WatchStock implements Serializable {

        @SerializedName("id")
        @Expose
        public String id;
        @SerializedName("stockid")
        @Expose
        public String stockid;
        @SerializedName("previousClose")
        @Expose
        public String previousClose;
        @SerializedName("marketopen")
        @Expose
        public String marketopen;
        @SerializedName("marketclose")
        @Expose
        public String marketclose;
        @SerializedName("sector")
        @Expose
        public String sector;
        @SerializedName("cryptocurrencyid")
        @Expose
        public String cryptocurrencyid;
        @SerializedName("cmc_rank")
        @Expose
        public String cmcRank;

        @SerializedName("crypto_logo")
        @Expose
        public String crypto_logo;

        @SerializedName("marketcapital")
        @Expose
        public String marketcapital;
        @SerializedName("supply")
        @Expose
        public String supply;
        @SerializedName("country_id")
        @Expose
        public String countryId;
        @SerializedName("marketId")
        @Expose
        public String marketId;
        @SerializedName("marketname")
        @Expose
        public String marketname;
        @SerializedName("currencyid")
        @Expose
        public String currencyid;
        @SerializedName("ask")
        @Expose
        public String ask;
        @SerializedName("bid")
        @Expose
        public String bid;

        @SerializedName("currency_netchange")
        @Expose
        public String currency_netchange;
        @SerializedName("currency_perchange")
        @Expose
        public String currency_perchange;
        @SerializedName("currency_volume")
        @Expose
        public String currency_volume;
        @SerializedName("crypto_market_id")
        @Expose
        public String crypto_market_id;
        @SerializedName("currency_market_id")
        @Expose
        public String currency_market_id;
        @SerializedName("currency_symbol")
        @Expose
        public String currency_symbol;

        @SerializedName("currency_firstflag")
        @Expose
        public String currency_firstflag;
         @SerializedName("currency_secondflag")
         @Expose
         public String currency_secondflag;

        @SerializedName("symbol")
        @Expose
        public String symbol;

        @SerializedName("image")
        @Expose
        public String image;
        @SerializedName("latestVolume")
        @Expose
        public String latestVolume;
        @SerializedName("latestPrice")
        @Expose
        public String latestPrice;
        @SerializedName("changePercent")
        @Expose
        public String changePercent;
        @SerializedName("companyName")
        @Expose
        public String companyName;


    }
}

File: app/src/main/java/stock/com/ui/pojo/WatchlistPojo.kt
package stock.com.ui.pojo

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

import java.io.Serializable
import java.util.ArrayList

class WatchlistPojo : BasePojo() {
    @SerializedName("stock")
    @Expose
    var stock: ArrayList<WatchStock>? = null

    inner class WatchStock : Serializable {

        @SerializedName("id")
        @Expose
        var id: String? = null
        @SerializedName("stockid")
        @Expose
        var stockid: String? = null
        @SerializedName("previousClose")
        @Expose
        var previousClose: String? = null
        @SerializedName("marketopen")
        @Expose
        var marketopen: String? = null
        @SerializedName("marketclose")
        @Expose
        var marketclose: String? = null
        @SerializedName("sector")
        @Expose
        var sector: String? = null
        @SerializedName("cryptocurrencyid")
        @Expose
        var cryptocurrencyid: String? = null
        @SerializedName("cmc_rank")
        @Expose
        var cmcRank: String? = null

        @SerializedName("crypto_logo")
        @Expose
        var crypto_logo: String? = null

        @SerializedName("marketcapital")
        @Expose
        var marketcapital: String? = null
        @SerializedName("supply")
        @Expose
        var supply: String? = null
        @SerializedName("country_id")
        @Expose
        var countryId: String? = null
        @SerializedName("marketId")
        @Expose
        var marketId: String? = null
        @SerializedName("marketname")
        @Expose
        var marketname: String? = null
        @SerializedName("currencyid")
        @Expose
        var currencyid: String? = null
        @SerializedName("ask")
        @Expose
        var ask: String? = null
        @SerializedName("bid")
        @Expose
        var bid: String? = null

        @SerializedName("currency_netchange")
        @Expose
        var currency_netchange: String? = null
        @SerializedName("currency_perchange")
        @Expose
        var currency_perchange: String? = null
        @SerializedName("currency_volume")
        @Expose
        var currency_volume: String? = null
        @SerializedName("crypto_market_id")
        @Expose
        var crypto_market_id: String? = null
        @SerializedName("currency_market_id")
        @Expose
        var currency_market_id: String? = null
        @SerializedName("currency_symbol")
        @Expose
        var currency_symbol: String? = null

        @SerializedName("currency_firstflag")
        @Expose
        var currency_firstflag: String? = null
        @SerializedName("currency_secondflag")
        @Expose
        var currency_secondflag: String? = null

        @SerializedName("symbol")
        @Expose
        var symbol: String? = null

        @SerializedName("image")
        @Expose
        var image: String? = null
        @SerializedName("latestVolume")
        @Expose
        var latestVolume: String? = null
        @SerializedName("latestPrice")
        @Expose
        var latestPrice: String? = null
        @SerializedName("changePercent")
        @Expose
        var changePercent: String? = null
        @SerializedName("companyName")
        @Expose
        var companyName: String? = null


    }
}
